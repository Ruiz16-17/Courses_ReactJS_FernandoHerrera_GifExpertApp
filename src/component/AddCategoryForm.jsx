import PropTypes from 'prop-types';
import React, { useState } from 'react';

export const AddCategoryForm = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const handleSubmit = (event) => {

        event.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(categories => [inputValue,...categories]);
            setInputValue('');
        }
        
    }

    return (

        <form onSubmit={handleSubmit}>

            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>

    )
}

AddCategoryForm.propType = {
    setCategories : PropTypes.func.isRequired
}