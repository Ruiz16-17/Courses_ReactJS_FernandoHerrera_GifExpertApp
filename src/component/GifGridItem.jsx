import PropTypes from 'prop-types';
import React from 'react';

export const GifGridItem = ({id, url, title}) => {
    return (
        <div 
            key={id}
            className="card animate__animated animate__bounce"
        >
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propType = {
    id : PropTypes.string.isRequired,
    title : PropTypes.string.isRequired,
    url :PropTypes.string.isRequired
    
}