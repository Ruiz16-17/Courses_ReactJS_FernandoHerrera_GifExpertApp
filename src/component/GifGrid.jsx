import PropTypes from 'prop-types';
import React from 'react';
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {loading, data: images} = useFetchGif(category);

    return (
        <>

            <h3 className="animate__animated animate__flash">{category}</h3>

            {loading && <p>Loading...</p> }

            <div className="card-grid">
            
                {images.map(image => {

                    return (
                        <GifGridItem
                            key={image.id}
                            {...image}

                        />
                    )
                })}

            </div>
        </>
    )
}

GifGrid.propType = {
    category : PropTypes.string.isRequired
    
}