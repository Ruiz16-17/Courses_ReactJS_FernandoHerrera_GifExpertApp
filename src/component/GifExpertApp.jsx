import React, { useState } from 'react';
import { AddCategoryForm } from './AddCategoryForm';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            
            <AddCategoryForm setCategories={setCategories}/>
            
            <hr/>

            <ol>
                {
                    categories.map(
                        category => {
                            return <GifGrid 
                                key={category}
                                category={category} />
                        }
                    )
                
                }
            </ol>

        </>
    )
}
