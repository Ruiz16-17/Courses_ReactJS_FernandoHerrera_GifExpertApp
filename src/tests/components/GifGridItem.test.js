import { shallow } from "enzyme";
import React from "react";
import { GifGridItem } from "../../component/GifGridItem";

describe('Test in <GifGridItem/>', () => {

    const id = 1;
    const title = 'Title GifGridItem';
    const url = 'https://localhost:8080/something.jpg';

    const image ={
        id: id,
        title: title,
        url: url
    }
    const wrapper = shallow(<GifGridItem {...image}/>)

    test('Should show correctly the component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Should have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    
    test('should have a the same image as a url and alt props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
        
    })
    
    test('should have animate__bounce', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes(className)).toBe(true);
    })
    
});
