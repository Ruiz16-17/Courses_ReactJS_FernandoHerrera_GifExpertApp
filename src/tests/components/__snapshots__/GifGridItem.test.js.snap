// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem/> Should show correctly the component 1`] = `
<div
  className="card animate__animated animate__bounce"
  key="1"
>
  <img
    alt="Title GifGridItem"
    src="https://localhost:8080/something.jpg"
  />
  <p>
    Title GifGridItem
  </p>
</div>
`;
